### PR TITLE
Add section on build environments and indexing

### PR DESCRIPTION
**Pull request by LisaFC**
Update for #653

### CODE REVIEW

**Comment by LisaFC** [comment]
@emckean @chalin how does this look?

**Comment by chalin** [comment]
Preview links to changed sections:
- https://deploy-preview-663--docsydocs.netlify.app/docs/adding-content/navigation/#configure-search-with-a-google-custom-search-engine
- https://deploy-preview-663--docsydocs.netlify.app/docs/deployment/#build-environments-and-indexing

File: userguide/content/en/docs/Deployment/_index.md [hunk 1]
```diff
@@ -29,6 +29,19 @@ If you've just added the theme as a submodule in a local version of your site an
...
+## Build environments and indexing
+
+By default, Hugo sites built with `hugo` (rather than served locally with `hugo server`) have the Hugo build environment `production`. Deployed Docsy sites with `production` builds can be indexed by search engines, including [Google Custom Search Engines](/docs/adding-content/navigation/#configure-search-with-a-google-custom-search-engine). Production builds also have optimized JavaScript and CSS for live deployment (for example, minified JS rather than the more legible original source).
+
+If you do not want your deployed site to be indexed by search engines (for example if you are still developing your live site), or if you want to build a development version of your site for offline analysis, you can set your Hugo build environment to something else such as `development` (the default for local deploys with `hugo server`), `test`, or another environment name of your choice.
+
+The simplest way to set this is by using the `-e` flag when specifying or running your `hugo` command, as in the following example:
+
+```
+hugo -e myenvironment
```
Comment (chalin) [inline userguide/content/en/docs/Deployment/_index.md:41]: nit: maybe use `-e staging` as an example instead?

**Review by chalin — APPROVED** [review]
LGTM, thanks!

For one small suggestion, see the inlined comment.

File: userguide/content/en/docs/Deployment/_index.md (hunk 1 above)
Comment (chalin) [reply userguide/content/en/docs/Deployment/_index.md:41]: Or maybe `-e development`?

Comment (LisaFC) [reply userguide/content/en/docs/Deployment/_index.md]: Done.

**Pull request merged**